crates/bevy_tasks/src/priority_executor.rs

```diff
@@ -0,0 +1,70 @@
+use std::future::Future;
+use std::sync::atomic::AtomicUsize;
+
+use async_executor::{Executor, Task};
+use event_listener::Event;
+use futures_lite::FutureExt;
+
+/// Task priority.
+#[repr(usize)]
```

Comment:
```suggestion
```
I don't believe this is necessary and can actually be a pessimization. 
from the [Rustonomicon](https://doc.rust-lang.org/nomicon/other-reprs.html) 
> Adding an explicit repr to an enum suppresses the null-pointer optimization.

You can safely cast this to a `uszie` with the `repr` :)